### ISSUE TITLE
Escape key creates cryptic characters 

### ISSUE THREAD

**Issue by hwalinga** [drmikehenry/vim-fixkey#10]
I am using vim 8.0 with urxvt 9.22. Your plugin seems to work flawlessly for alt mappings. However, it broke my escape key. Whenever I press Escape, it seems I am in normal mode, but the next key press will produce a special/cryptic character, like é or í. Do you have any idea how to fix this? 

EDIT: I also have tried various versions of term. "screen-256color, rxvt, rxvt-unicode, rxvt-unicode-256color, rxvt-256color". Al didn't seem to work. I also made sure I load this plugin after I set the term variable.

**Comment by drmikehenry** [comment]
I suspect you may need to adjust your setting of `ttimeoutlen` (which controls the timeout that determines whether consecutive bytes represent separate keypresses or a single multi-byte keypress).  By default, Vim sets `ttimeoutlen` to `-1`, meaning it tracks the value of `timeoutlen`.  Fixkey typically changes `ttimeoutlen` to 5 milliseconds, but it makes that adjustment only if it finds `ttimeoutlen` is set to its default (`-1`) value.  If you have a value of `ttimeoutlen` that is too large, Vim can treat two separate keypresses as if they were a single multi-byte keypress.  You can see your current value of `ttimeoutlen` via:
```viml
  set ttimeoutlen?
```

You should see a value of `5` assuming you have the Fixkey plugin active and you haven't changed the value of `ttimeoutlen` (either directly or indirectly through another plugin).

You can demonstrate the problem by intentionally setting the value too high; for example, try changing to a one-second timeout via:
```viml
  :set ttimeoutlen=1000
```

Then enter insert mode via `i`.  If you try to exit and immediately re-enter insert mode by pressing `Escape` immediately followed by `i`, Vim will treat the two keystrokes as the character `é`.

You can try reducing the value of `ttimeoutlen` to as little as 1 millisecond to see if this fixes the problem.  For example:
```viml
set ttimeoutlen=1
```
This should fix the problem unless you type another character very soon after pressing `Escape`.  If this doesn't fix the problem because you such a fast typist (in which case, congratulations on your typing skill :-)), the only remedy I can offer is intentionally slowing down a bit after pressing `Escape`.

**Comment by hwalinga** [comment]
Thanks for your explanation. The problem was a lot more complex, but you gave me enough hints to solve it. 

I had made mappings for certain key combinations myself before I discovered this plugin. I made them by prepending them with CTRL-V. However, vim interpreted this as a key chord mapping (like mapping kj to esc, between k and j now needs a certain amount of time that is specified by timeoutlen (not ttimeoutlen)). So what happend, my home made mappings were interpreted as those key chords, making vim wait timeoutlen instead of ttimeoutlen whenever I would type escape, before sending the keycodes, which for me (and most people) is much longer. 

Solving this:

:imap <Esc> should return nothing. All mappings returned by this need to be removed, otherwise timeoutlen is used by <Esc> instead of ttimeoutlen.
